mono/metadata/w32file-win32.c

```diff
@@ -370,6 +370,8 @@ mono_w32file_get_disk_free_space (const gunichar2 *path_name, guint64 *free_byte
 	return result;
 }
 
+G_BEGIN_DECLS // FIXMEcxx wasm
```

Comment:
Oh, so wasm is using win32 now?

Reply:
mono_w32file_get_volume_information "has to" be externC, so I marked them all.
"has to" is a probably exaggeration though and could be avoided.

Reply:
How about you mark it in the header like all other functions that need to be marked as `extern "C"`?

Reply:
"has to" because I was getting all lanes to compile and link and pass all tests.  A bit rushed perhaps. wasm was the only lane with failures, non-obvious test failures, but worked compiled as C, and sprinkling externC around fixed it. Some of the need for externC is obvious but some is not.
It occurs here: https://github.com/mono/mono/blob/master/mono/mini/mini-wasm.c#L580
Yes marking only in header should suffice. I'll try that.